website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -360,13 +370,150 @@ assert_eq!(.tags.environment, "production")
+[transforms.add_host_metadata]
+type = "remap"
+inputs = ["sanitize"]
+source = '''
+.tags.host = "web-backend1.vector-user.biz"
+'''
+
+[[tests]]
+name = "Multiple chained remap transforms"
+
+[[tests.inputs]]
+type = "log"
+# Insert test input events into the first transform
+insert_at = "add_env_metadata"
+
+# The input event to insert into the first transform in the chain
+[tests.inputs.log_fields]
+message = "image successfully uploaded"
+code = 202
+username = "tonydanza1337"
+email = "tony@whostheboss.com"
+transaction_id = "bcef6a6a-2b72-4a9a-99a0-97ae89d82815"
+
+[[tests.outputs]]
+# Extract test outputs from the last transform
+extract_from = "add_host_metadata"
+
+[[tests.outputs.conditions]]
+type = "vrl"
+# Our VRL assertions for the test output
+source = '''
+assert_eq!(.tags.environment, "production", "incorrect environment tag")
+assert_eq!(.tags.host, "web-backend1.vector-user.biz", "incorrect host tag")
+assert!(!exists(.username))
+assert!(!exists(.email))
+
+valid_transaction_id = exists(.transaction_id) &&
+  is_string(.transaction_id) &&
+  length!(.transaction_id) == 36
+
+assert!(valid_transaction_id, "transaction ID invalid")
+'''
+```
+
+From a testing standpoint, all three transforms here can be thought of as a single unit. One example
+event is inserted at the beginning of the chain (`add_env_metadata`), one output test event is
+extracted from the end of the chain (`add_host_metadata`), and one set of VRL
+[assertions](#assertions) verifies that that output event conforms to our expectations.
+
+You could also test a subset of this transform chain. This configuration, for example, would test
+only the first two transforms (`add_env_metadata` and `sanitize`):
+
+```toml
+[[tests]]
+name = "First two transforms"
+
+[[tests.inputs]]
+type = "log"
+# Insert test input into the first transform
+insert_at = "add_env_metadata"
+
+# For comparison, we can use the same input event as above
+[tests.inputs.log_fields]
+message = "image successfully uploaded"
+code = 202
+username = "tonydanza1337"
+email = "tony@whostheboss.com"
+transaction_id = "bcef6a6a-2b72-4a9a-99a0-97ae89d82815"
+
+[[tests.outputs]]
+# Extract test output from the second transform rather than the last
+extract_from = "sanitize"
+
+[[tests.outputs.conditions]]
+type = "vrl"
+source = '''
+assert_eq!(.tags.environment, "production", "incorrect environment tag")
+assert!(!exists(.tags.host), "host tag included")
+assert!(!exists(.username))
+assert!(!exists(.email))
+
+valid_transaction_id = exists(.transaction_id) &&
+  is_string(.transaction_id) &&
+  length!(.transaction_id) == 36
+
+assert!(valid_transaction_id, "transaction ID invalid")
+'''
+```
+
+In the VRL conditions for this two-transform test, notice that the assertion regarding the `host`
+tag is changed to this, which verifies that that tag isn't present, which is what we should expect
+given that the `add_host_metadata` transform isn't included here:
+
+```text
```

Comment:
Jean usually uses coffeescript as the highlighting for raw VRL which looks reasonable, btw

Reply:
In this case I actually don't want it to stand out, as in the example immediately above there's no highlighting for the VRL assertions.